### DISCUSSION TITLE
split-dwarf: Object file contain directory of cached paths for .dwo file

### DISCUSSION THREAD

**Discussion by dyadav7** [ccache/ccache#591] [Support]
### Description ###
Split-dwarf is enabled. The readelf command on object files show the DW_AT_comp_dir field and this field has other user directory paths. 

Can ccache modify the files such that DW_AT_comp_dir field show the current user directory  ?

DW_AT_comp_dir below show user1 and user2.

    $ readelf -wi dir1/file11.o | grep /
    readelf: Warning: skipping unexpected symbol type TLS in section .rela.debug_addr relocation 809
        <1c>   DW_AT_GNU_dwo_name: (indirect string, offset: 0x1d): dir1/file11.dwo
        <20>   DW_AT_comp_dir    : (indirect string, offset: 0x0): /home/user1/src
    $ readelf -wi dir2/file22.o | grep /
        <1c>   DW_AT_GNU_dwo_name: (indirect string, offset: 0x20): dir2/file22.dwo
        <20>   DW_AT_comp_dir    : (indirect string, offset: 0x0): /home/user2/src
    $

**Comment by jrosdahl** [comment]
> Can ccache modify the files such that DW_AT_comp_dir field show the current user directory ?

I don't think that it's reasonable for ccache to rewrite object files.

But you should never get a reference to another user's compilation directory in the object file unless you have chosen to to set `hash_dir = false`. I guess that's the case here? See  https://ccache.dev/manual/3.7.9.html#_compiling_in_different_directories for more information.

**Comment by dyadav7** [comment]
Yes, hash_dir is false.

**Comment by jrosdahl** [comment]
Can you use the `-fdebug-prefix-map=...` option instead?

Are you happy with the answer so that we can close the issue?

**Comment by dyadav7** [comment]
I have a different solution by executing "dwp" (dwp - The DWARF packaging utility part of binutils 2.34) after the compile/linking is complete. 
  If dwp executable is used on the executable/libraries then it reads all .dwo files and generates a .dwp file. The .dwp file always works fine. So gdb will read the .dwp file inplace of .dwo files now.
  Only issue is that it forces user to use dwp now (it may take some time on big executable).
  i am using dwp now.

So, the ticket can be closed.

I will check -fdebug-prefix-map=... later .
